app/main/forms.py

```diff
@@ -70,3 +70,7 @@ class InviteForm(DmForm):
         DataRequired(message="Email cannot be empty"),
         Email(message="Please enter a valid email address")
     ])
+
+
+class SellerUpdateCategoryPriceForm(DmForm):
+    new_supplier_category_price = StringField('New Price', validators=[InputRequired('Please enter a valid price')])
```

Comment:
Did you try implement a [custom validator](https://wtforms.readthedocs.io/en/2.3.x/validators/#custom-validators)?  You could try import your function and use it like this.  The function will need to return a `ValidationError`.
```suggestion
    new_supplier_category_price = StringField('New Price', validators=[InputRequired('Please enter a price'), is_float])
```

Reply:
custom validation added